include/seahorn/Transforms/Instrumentation/CrabAnalysisPass.hh

```diff
@@ -0,0 +1,37 @@
+#ifndef _CRAB_ANALYSIS_HH_
+#define _CRAB_ANALYSIS_HH_
+
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Pass.h"
+
+namespace clam {
+class CrabBuilderManager;
+class InterGlobalClam;
+} // namespace clam
+
+namespace seahorn {
+using namespace llvm;
+
+class CrabAnalysisPass : public ModulePass {
```

Comment:
Don't set `CrabAnalysisPass` as a LLVM pass. 
Since a while we try to rely on the pass manager as little as possible. 
Most of the code here is still fine. The only changes is to remove is `public Module Pass` and `getAnalysisUsage`. Then, the caller of this code (`CrabLowerIsDeref.cc`) which needs  to be a LLVM pass it will make sure that all the requirements of `CrabAnalysisPass` (if any) are fullfilled.

Reply:
Yes. I refactor this class to remove the LLVM pass.